pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -134,7 +135,10 @@ func discoverAPI(client kubernetes.Interface, kindArg string) (kind string, grou
 
 	resourceLists, err := client.Discovery().ServerPreferredNamespacedResources()
 	if err != nil {
-		return "", nil, fmt.Errorf("failed to discover apigroup for kind %q: %v", kind, err)
+		if !discovery.IsGroupDiscoveryFailedError(err) {
+			return "", nil, fmt.Errorf("failed to discover preferred resources: %v", err)
+		}
+		glog.Warningf("failed to discover some groups: %v", err.(*discovery.ErrGroupDiscoveryFailed).Groups)
```

Comment:
nit: Failed

Reply:
Could we simply do `glog.Warningf("Failed to discover some groups: %v", err)`?

Reply:
Yes, makes sense. With this change, it logs:

```
W0528 06:29:45.895057   66600 k8sclient.go:142] Failed to discover some groups: unable to retrieve the complete list of server APIs: custom.group.example.com/v1alpha1: the server is currently unable to handle the request
```

instead of (earlier):

```
W0528 06:29:56.458506   66731 k8sclient.go:141] failed to discover some groups: map[custom.group.example.com/v1alpha1:the server is currently unable to handle the request]
```